### ISSUE TITLE
Serve books using Karax

### ISSUE THREAD

**Issue by Clonkk** [pietroppeter/nimib#46]
It would be really useful to integrate Karax to Nimib to serve documentation - like ``mdbook serve`` - it would make it easy to deploy as well and not reliant upon github page.

What do you think ?

**Comment by pietroppeter** [comment]
well, if the output is static files and the issue is to decouple from github pages probably something simple like https://github.com/h3rald/nimhttpd could work.

In general I think both jester (for serving stuff) and karax (for SPA and interactivity in the page and also working alongside jester to serve stuff) could be very nice to have integrated in a nimib workflow (one could be something closer to a jupyter notebook with that, or even better something like https://streamlit.io/).

For the moment I think I will concentrate on fixing stuff and on the api for the "static" use case (with target use cases a blog, a site like adventofnim, or a book like scinim/getting-started), as you have seen there is lots to do on that side. Feel free to experiment anyway. :)

**Comment by Clonkk** [comment]
> For the moment I think I will concentrate on fixing stuff and on the api for the "static" use case (with target use cases a blog, a site like adventofnim, or a book like scinim/getting-started), as you have seen there is lots to do on that side. Feel free to experiment anyway. :)

Yes, this was just an idea to keep on the side for later. Also, I don't know which http server whould be best, I only know Karax and Prologue (which is probably overkill here) which is why I suggested Karax :)

Definitly not a priority

**Comment by pietroppeter** [comment]
Renamed issue (karax is not a http server). To encapsulate better the enhancement request, the main features that [mdbook serve](https://rust-lang.github.io/mdBook/cli/serve.html) provides are:

* serve static files (like any web server such as httpd, independent from nimib)
* watch for changes and relaunch the build process (this depends on nimib build process)

This looks like a good addition for nimib, so for me it goes in the roadmap.
